### PR TITLE
GA Migration Java PostCommit Validate Runners (Spark, Spark Java 11, SparkStructuredStream, ULR) 

### PR DESCRIPTION
**Pull request by MarcoRob**
As part of the migration of Precommit and Postcommit Jobs from Jenkins to GA in self-hosted runners, this PR contains:

- Migrated workflow [job-postcommit-java-vr-spark.yml](https://github.com/apache/beam/blob/492e89637fef60044d23a09d43af2693fc6e2abc/.github/workflows/job-postcommit-java-vr-spark.yml)
- Migrated workflow [job-postcommit-java-vr-spark-java11.yml](https://github.com/apache/beam/blob/492e89637fef60044d23a09d43af2693fc6e2abc/.github/workflows/job-postcommit-java-vr-spark-java11.yml)
- Migrated workflow [job-postcommit-java-vr-spark-structured-streaming.yml](https://github.com/apache/beam/blob/492e89637fef60044d23a09d43af2693fc6e2abc/.github/workflows/job-postcommit-java-vr-spark-structured-streaming.yml)
- Migrated workflow [job-postcommit-java-vr-ulr.yml](https://github.com/apache/beam/blob/492e89637fef60044d23a09d43af2693fc6e2abc/.github/workflows/job-postcommit-java-vr-ulr.yml)

The migrated workflows were removed from [.test-infra/jenkins/README.md](https://github.com/apache/beam/blob/c1f2e687909542ae20c333ba507e020f1273f78d/.test-infra/jenkins/README.md) and added to [CI.md](https://github.com/apache/beam/blob/492e89637fef60044d23a09d43af2693fc6e2abc/CI.md)

`DO NOT MERGE` until the effort to use self-hosted runners is completed https://github.com/apache/beam/pull/22703

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

File: .github/workflows/job-postcommit-java-vr-spark-java11.yml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        uses: actions/checkout@v3
+        with:
+          persist-credentials: false
+      - name: Setup environment
+        uses: ./.github/actions/setup-self-hosted-action
+      - name: Spark 3 Test Jobs
+        shell: bash
+        run: |
+          ./gradlew :runners:spark:3:jar
+          sleep 5
```
Comment (damccorm) [inline .github/workflows/job-postcommit-java-vr-spark-java11.yml:54]: Why do we need a sleep statement here?

**Review by damccorm — COMMENTED** [review]
Just had one question

File: .github/workflows/job-postcommit-java-vr-spark-java11.yml (hunk 1 above)
Comment (MarcoRob) [reply .github/workflows/job-postcommit-java-vr-spark-java11.yml:54]: It is just a precaution to wait for the next command to be run, do you think we should eliminate it? @damccorm

Comment (MarcoRob) [reply .github/workflows/job-postcommit-java-vr-spark-java11.yml:54]: Is there any comment? Do I remove the `sleep` ?

Comment (damccorm) [reply .github/workflows/job-postcommit-java-vr-spark-java11.yml:54]: Sorry for the slow response here - could you help me understand why we need this precaution? Will this command not block until the spark server is spun up?

Comment (MarcoRob) [reply .github/workflows/job-postcommit-java-vr-spark-java11.yml:54]: Yes, I tested again and at the beginning, the `sleep` it was a precaution to give time to run the first command and then the next one, but testing out the `sleep` is not necessary as you mentioned, I will remove it.

Comment (MarcoRob) [reply .github/workflows/job-postcommit-java-vr-spark-java11.yml]: Fix applied

**Review by damccorm — APPROVED** [review]

**Pull request closed without merging**